app/src/main/java/com/nikunjsondagar/composedemo/ui/presentation/SharedViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.nikunjsondagar.composedemo.ui.presentation
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+
+class SharedViewModel : ViewModel() {
+
+    private val _searchViewState: MutableState<SearchViewState> =
```

Comment:
I am not big fan of using _ for private field. How about using searchViewMutableState for private and searchViewState for public?